exchanges/gateio/gateio_wrapper.go

```diff
@@ -534,40 +556,79 @@ func (g *Gateio) GetFeeByType(feeBuilder *exchange.FeeBuilder) (float64, error)
 
 // GetActiveOrders retrieves any orders that are active/open
 func (g *Gateio) GetActiveOrders(req *order.GetOrdersRequest) ([]order.Detail, error) {
+	var orders []order.Detail
 	var currPair string
 	if len(req.Currencies) == 1 {
 		currPair = req.Currencies[0].String()
 	}
+	if g.Websocket.CanUseAuthenticatedWebsocketForWrapper() {
+		for i := 0; ; i += 100 {
+			resp, err := g.wsGetOrderInfo(req.OrderType.String(), i, 100)
+			if err != nil {
+				return orders, err
+			}
 
-	resp, err := g.GetOpenOrders(currPair)
-	if err != nil {
-		return nil, err
-	}
-
-	var orders []order.Detail
-	for i := range resp.Orders {
-		if resp.Orders[i].Status != "open" {
-			continue
+			for j := range resp.WebSocketOrderQueryRecords {
+				orderSide := order.Buy
+				if resp.WebSocketOrderQueryRecords[j].Type == 1 {
+					orderSide = order.Sell
+				}
+				orderType := order.Market
+				if resp.WebSocketOrderQueryRecords[j].OrderType == 1 {
+					orderType = order.Limit
+				}
+				firstNum, decNum, err := convert.SplitFloatDecimals(resp.WebSocketOrderQueryRecords[j].Ctime)
+				if err != nil {
+					return orders, err
+				}
+				orderDate := time.Unix(firstNum, decNum)
+				orders = append(orders, order.Detail{
+					Exchange:        g.Name,
+					AccountID:       strconv.FormatInt(resp.WebSocketOrderQueryRecords[j].User, 64),
```

Comment:
These should be base 10?

Reply:
https://play.golang.org/p/g2w1N7VNtry indeed, looks like it will cause a panic

Reply:
Oh my goodness. Good catch!

Reply:
🦅